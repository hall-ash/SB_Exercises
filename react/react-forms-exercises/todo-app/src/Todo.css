.Todo-item {
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.Todo-item button {
  margin-left: 10px;
}