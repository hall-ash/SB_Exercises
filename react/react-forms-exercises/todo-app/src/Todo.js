import React, {useState} from 'react';
import './Todo.css';

const Todo = ({name, deleteTodo, editTodo, key}) => {
  const [newName, setNewName] = useState(name);
  const [beingEdited, setBeingEdited] = useState(false);

  const handleChange = evt => setNewName(evt.target.value);

  const toggleEdit = () => {
    setBeingEdited(edit => !edit);
  };

  const handleEdit = evt => {
    evt.preventDefault();
    editTodo(key, newName);
    toggleEdit();
  };

  return beingEdited ? 
    <div>
      <form onSubmit={handleEdit}>
        <input type="text" value={newName} onChange={handleChange} />
        <button>Save</button>
      </form>
    </div> :
   <div className="Todo-item">
      <li>{newName}</li>
      <button onClick={toggleEdit}>Edit</button>
      <button onClick={deleteTodo}>X</button>
   </div>
  ;
};

export default Todo;