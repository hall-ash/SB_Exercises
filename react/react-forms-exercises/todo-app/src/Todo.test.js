import React from 'react';
import { render } from '@testing-library/react';
import Todo from './Todo';

test('renders without crashing', () => {
  render(<Todo />);
});

test('matches snapshot', () => {
  const { asFragment } = render(<Todo />)
  expect(asFragment()).toMatchSnapshot();
});