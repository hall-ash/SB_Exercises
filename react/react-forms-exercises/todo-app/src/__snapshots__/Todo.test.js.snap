// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo-item"
  >
    <li />
    <button>
      Edit
    </button>
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
